tests/it/arrow.rs

```diff
@@ -0,0 +1,350 @@
+        offsets.into(),
+        Box::new(values.clone()),
+        Some(validity),
+    );
+
+    test_conversion(&list);
+
+    let list = ListArray::<i32>::new_empty(data_type);
+    test_conversion(&list);
+
+    let validity = [true, true, false, false, true].into_iter().collect();
+    let offsets = Offsets::try_from_iter(vec![0, 2, 2, 2, 0]).unwrap();
+    let data_type = DataType::LargeList(Box::new(Field::new("element", DataType::Utf8, true)));
+    let list = ListArray::<i64>::new(
+        data_type.clone(),
+        offsets.into(),
+        Box::new(values),
+        Some(validity),
+    );
+
+    test_conversion(&list);
+
+    let list = ListArray::<i64>::new_empty(data_type);
+    test_conversion(&list);
+}
+
+#[test]
+fn test_list_struct() {
+    let values = make_struct();
+    let validity = [true, true, false, true].into_iter().collect();
+    let offsets = Offsets::try_from_iter(vec![0, 1, 0, 2]).unwrap();
+    let list = ListArray::<i32>::new(
+        DataType::List(Box::new(Field::new(
+            "element",
+            values.data_type().clone(),
+            true,
+        ))),
+        offsets.into(),
+        Box::new(values),
+        Some(validity),
+    );
+
+    test_conversion(&list);
+}
+
+#[test]
+fn test_dictionary() {
+    let nulls = [true, false, true, true, true].into_iter().collect();
+    let keys = PrimitiveArray::new(DataType::Int16, vec![1_i16, 1, 0, 2, 2].into(), Some(nulls));
+    let values = make_struct();
+    let dictionary = DictionaryArray::try_new(
+        DataType::Dictionary(
+            IntegerType::Int16,
+            Box::new(values.data_type().clone()),
+            false,
+        ),
+        keys,
+        Box::new(values),
+    )
+    .unwrap();
+
+    test_conversion(&dictionary);
+}
+
+#[test]
+fn test_fixed_size_binary() {
+    let data = (0_u8..16).collect::<Vec<_>>();
+    let nulls = [false, false, true, true, true, false, false, true]
+        .into_iter()
+        .collect();
+
+    let array = FixedSizeBinaryArray::new(DataType::FixedSizeBinary(2), data.into(), Some(nulls));
+    test_conversion(&array);
+}
+
+#[test]
+fn test_fixed_size_list() {
+    let values = vec![1_i64, 2, 3, 4, 5, 6, 7, 8];
+    let nulls = [false, false, true, true, true, true, false, false]
+        .into_iter()
+        .collect();
+    let values = PrimitiveArray::new(DataType::Int64, values.into(), Some(nulls));
+
+    let nulls = [true, true, false, true].into_iter().collect();
+    let array = FixedSizeListArray::new(
+        DataType::FixedSizeList(Box::new(Field::new("element", DataType::Int64, true)), 2),
+        Box::new(values),
+        Some(nulls),
+    );
+
+    test_conversion(&array);
+}
+
+#[test]
+fn test_map() {
+    let keys = Utf8Array::<i32>::from_iter(
+        ["key1", "key2", "key3", "key1", "key2"]
+            .into_iter()
+            .map(Some),
+    );
+    let values = PrimitiveArray::<i32>::from_iter([Some(1), None, Some(3), Some(1), None]);
```

Comment:
is it intended that `key1 = Some(1)` is repeated? Would it be adding more coverage if the second value was something different  (like `key1 = Some(10)`)?